game-core/src/main/java/games/strategy/engine/data/MoveDescription.java

```diff
@@ -26,6 +26,9 @@ public MoveDescription(
       final Map<Unit, Collection<Unit>> dependentUnits) {
     super(Collections.unmodifiableCollection(units));
     this.route = Preconditions.checkNotNull(route);
+    if (route.hasNoSteps()) {
+      throw new IllegalStateException("Invalid route: " + route);
+    }
```

Comment:
Guavas `Preconditions#checkState` makes this a one-liner

Reply:
Done